### PR TITLE
[11.0][MIG] web_switch_context_warning

### PR DESCRIPTION
**Pull request by AdriaGForgeFlow**
Standard migration of module web_switch_context_warning to v11 from #1287 

cc ~ @Eficent

### CODE REVIEW

**Comment by pedrobaeza** [comment]
A bit of commit squashing is recommended

File: web_switch_company_warning/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2017 Akretion <raphael.reverdy@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Multicompany - Switch Company Warning",
+    "summary": "Show a warning if current company has been switched"
+               " in another tab or window.",
+    "version": "11.0.1.0.0",
+    "category": "web",
+    "website": "http://akretion.com",
```
Comment (tbaden) [inline web_switch_company_warning/__manifest__.py]: Please use the website https://github.com/OCA/web

File: web_switch_company_warning/static/src/js/switch_company_warning.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+odoo.define('web_switch_company_warning.widget', function (require) {
+    'use strict';
+
+    var Widget = require('web.Widget');
+    var UserMenu = require('web.UserMenu');
+    var session = require('web.session');
+    //Show a big banner in the top of the page if the company has been
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning.js]: [W7903(javascript-lint), ] Expected space or tab after '//' in comment. [Error/spaced-comment]

File: web_switch_company_warning/static/src/js/switch_company_warning.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    //changed in another tab or window (in the same browser)
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning.js]: [W7903(javascript-lint), ] Expected space or tab after '//' in comment. [Error/spaced-comment]

File: web_switch_company_warning/static/src/js/switch_company_warning.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    if (!window.SharedWorker) {
+        //not supported
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning.js]: [W7903(javascript-lint), ] Comments should not begin with a lowercase character. [Error/capitalized-comments]
[W7903(javascript-lint), ] Expected space or tab after '//' in comment. [Error/spaced-comment]

File: web_switch_company_warning/static/src/js/switch_company_warning.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        return;
+    }
+    var SwitchCompanyWarningWidget = Widget.extend({
+        template:'web_switch_company_warning.warningWidget',
+        init: function() {
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning.js]: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]

File: web_switch_company_warning/static/src/js/switch_company_warning.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            this._super();
+            var self = this;
+            var w = new SharedWorker('/web_switch_company_warning/static/src/js/switch_company_warning_worker.js');
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning.js]: W7903(javascript-lint), ] Line 19 exceeds the maximum line length of 80. [Error/max-len]

File: web_switch_company_warning/static/src/js/switch_company_warning.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+            w.port.addEventListener('message', function (msg) {
+                if (msg.data.type !== 'newCtx') {
+                    return;
+                }
+                if(msg.data.newCtx === self.generateSignature()) {
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning.js]: [W7903(javascript-lint), ] Expected space(s) after "if". [Error/keyword-spacing]

File: web_switch_company_warning/static/src/js/switch_company_warning.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+                    self.$el.hide();
+                } else {
+                    self.$el.show();
+                }
+            });
+            w.port.start();
+            w.port.postMessage(this.generateSignature());
+        },
+        generateSignature: function() {
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning.js]: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]

File: web_switch_company_warning/static/src/js/switch_company_warning.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+            return [session.company_id, session.db].join();
+        }
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning.js]: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]

File: web_switch_company_warning/static/src/js/switch_company_warning.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    });
+
+    UserMenu.include({
+        init: function(parent) {
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning.js]: [W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]

File: web_switch_company_warning/static/src/js/switch_company_warning.js [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+            this._super(parent);
+            var switchCompanyWarning = new SwitchCompanyWarningWidget();
+            // Choose where to append depending on whether web_responsive is installed or not
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning.js]: [W7903(javascript-lint), ] Line 42 exceeds the maximum line length of 80. [Error/max-len]

File: web_switch_company_warning/static/src/js/switch_company_warning.js [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+            if (document.getElementById('oe_main_menu_navbar')) {
+                switchCompanyWarning.appendTo('#oe_main_menu_navbar');
+            } else {
+                switchCompanyWarning.insertAfter('.main-nav');
+            }
+        }
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning.js]: [W7903(javascript-lint), ] Missing trailing comma. [Error/comma-dangle]

File: web_switch_company_warning/static/src/js/switch_company_warning_worker.js [hunk 13]
```diff
@@ -0,0 +1,22 @@
+"use strict";
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning_worker.js]: [W7903(javascript-lint), ] Use the function form of 'use strict'. [Error/strict]

File: web_switch_company_warning/static/src/js/switch_company_warning_worker.js [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+//Show a big banner in the top of the page if the company has been
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning_worker.js]: [W7903(javascript-lint), ] Expected space or tab after '//' in comment. [Error/spaced-comment]

File: web_switch_company_warning/static/src/js/switch_company_warning_worker.js [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+//changed in another tab or window (in the same browser)
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning_worker.js]: [W7903(javascript-lint), ] Expected space or tab after '//' in comment. [Error/spaced-comment]

File: web_switch_company_warning/static/src/js/switch_company_warning_worker.js [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+
+var con = [];
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning_worker.js]: [W7903(javascript-lint), ] Implicit global variable, assign as global property instead. [Error/no-implicit-globals]

File: web_switch_company_warning/static/src/js/switch_company_warning_worker.js [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+var lastCtx = null;
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning_worker.js]: [W7903(javascript-lint), ] Implicit global variable, assign as global property instead. [Error/no-implicit-globals]

File: web_switch_company_warning/static/src/js/switch_company_warning_worker.js [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+
+addEventListener("connect", function(ee) {
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning_worker.js]: [W7903(javascript-lint), ] Use the function form of 'use strict'. [Error/strict]
[W7903(javascript-lint), ] Missing space before function parentheses. [Error/space-before-function-paren]

File: web_switch_company_warning/static/src/js/switch_company_warning_worker.js [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+    var port = ee.ports[0];
+    con.push(port);
+
+    port.onmessage = function (e) {
+        var newCtx = e.data;
+
+        if (lastCtx && newCtx !== lastCtx) {
+            con.forEach(function (eport) {
+                eport.postMessage({type: "newCtx", "newCtx": newCtx, "lastCtx": lastCtx});
```
Comment (tbaden) [inline web_switch_company_warning/static/src/js/switch_company_warning_worker.js]: [W7903(javascript-lint), ] Line 17 exceeds the maximum line length of 80. [Error/max-len]

**Review by tbaden — CHANGES_REQUESTED** [review]
thanks for the work 👍 
I did a code review: here some changes.
for js it's the easiest way to use eslint with the oca [configuration](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#4javascript)

**Review by LoisRForgeFlow — APPROVED** [review]
Tested on EE version, :+1:

**Comment by LoisRForgeFlow** [comment]
@tbaden all your comments were attended, could you please update your review?

@pedrobaeza are you ok with the renaming done here?

**Comment by pedrobaeza** [comment]
Not sure if `context` is the best word. Maybe `environment`? But I don't have a strong opinion here, so proceed in the best way for you.

**Review by tarteo — APPROVED** [review]

**Pull request merged**